operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -416,4 +485,272 @@ private static Uri uri(FHIROperationContext operationContext, String uriPath) {
+        } else if (compartmentType.equals(Reference.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "authenticator", Organization.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "authenticator", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "author", Organization.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "author", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "custodian", Organization.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "subject", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+        } else if (compartmentType.equals(RequestGroup.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "author", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "participant", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+        } else if (compartmentType.equals(RiskAssessment.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "performer", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+        } else if (compartmentType.equals(Schedule.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "actor", Location.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "actor", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+        } else if (compartmentType.equals(ServiceRequest.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "performer", Organization.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "performer", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "requester", Organization.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "requester", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
+            addSearchParameterIfNotExcluded(compartmentType, "subject", Location.class.getSimpleName(), typeParameters, searchParameters); 
+        } else if (compartmentType.equals(Specimen.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentType, "collector", Practitioner.class.getSimpleName(), typeParameters, searchParameters);
```

Comment:
I don't think this one should be added. subject makes sense but collector does not